messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/DiscoveryGroupRemove.java

```diff
@@ -41,22 +39,34 @@
  */
 public class DiscoveryGroupRemove extends AbstractRemoveStepHandler {
 
-    public static final DiscoveryGroupRemove INSTANCE = new DiscoveryGroupRemove();
+    public static final DiscoveryGroupRemove INSTANCE = new DiscoveryGroupRemove(false);
+    public static final DiscoveryGroupRemove LEGACY_INSTANCE = new DiscoveryGroupRemove(true);
 
-    private DiscoveryGroupRemove() {
+    private final boolean isLegacyCall;
+
+    private DiscoveryGroupRemove(boolean isLegacyCall) {
         super();
+        this.isLegacyCall = isLegacyCall;
     }
 
     @Override
-    protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
-        PathAddress address = context.getCurrentAddress();
-
-        ModelNode model = resource.getModel();
-        if (JGROUPS_CLUSTER.resolveModelAttribute(context, model).isDefined() && !JGROUPS_CHANNEL.resolveModelAttribute(context, model).isDefined()) {
-            context.deregisterCapabilityRequirement(ClusteringDefaultRequirement.COMMAND_DISPATCHER_FACTORY.getName(), CAPABILITY.getDynamicName(address));
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        if (!isLegacyCall) {
+            ModelNode op = operation.clone();
+            PathAddress target = context.getCurrentAddress().getParent();
+            OperationStepHandler removeHandler;
+            try {
+                context.readResourceFromRoot(target.append(CommonAttributes.JGROUP_DISCOVERY_GROUP, context.getCurrentAddressValue()));
+                target = target.append(CommonAttributes.JGROUP_DISCOVERY_GROUP, context.getCurrentAddressValue());
+                removeHandler = JGroupDiscoveryGroupRemove.LEGACY_INSTANCE;
+            } catch(Resource.NoSuchResourceException ex) {
+                target = target.append(CommonAttributes.SOCKET_DISCOVERY_GROUP, context.getCurrentAddressValue());
+                removeHandler = SocketDiscoveryGroupRemove.LEGACY_INSTANCE;
+            }
+            op.get(OP_ADDR).set(target.toModelNode());
+            context.addStep(op, removeHandler, OperationContext.Stage.MODEL, true);
```

Comment:
Same question.  Why use an intermediate step?  Can we not simply use the non-legacy JGroups/Socket remove handler here?